src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageQueryEndpoint.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorLanguageQueryEndpoint : IRazorLanguageQueryHandler
+    {
+        public bool MutatesSolutionState => false;
+
+        private readonly RazorDocumentMappingService _documentMappingService;
+
+        public RazorLanguageQueryEndpoint(RazorDocumentMappingService documentMappingService)
+        {
+            _documentMappingService = documentMappingService;
+        }
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(RazorLanguageQueryParams request)
+        {
+            return new TextDocumentIdentifier
+            {
+                Uri = request.Uri
+            };
+        }
+
+        public async Task<RazorLanguageQueryResponse> HandleRequestAsync(RazorLanguageQueryParams request, RazorRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            requestContext.RequireDocumentContext();
+            var documentContext = requestContext.DocumentContext;
```

Comment:
This is old code right? Its `GetRequiredDocumentContext` now or something, you used above?